PurchasesCoreSwift/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -0,0 +1,506 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PurchasesOrchestrator.swift
+//
+//  Created by Andrés Boedo on 10/8/21.
+
+import Foundation
+import StoreKit
+
+public typealias PurchaseCompletedBlock = (SKPaymentTransaction?, PurchaserInfo?, Error?, Bool) -> Void
+public typealias RCDeferredPromotionalPurchaseBlock = (@escaping PurchaseCompletedBlock) -> Void
+
+@objc(RCPurchasesOrchestratorDelegate) public protocol PurchasesOrchestratorDelegate {
+    func shouldPurchasePromoProduct(_ product: SKProduct, defermentBlock: @escaping RCDeferredPromotionalPurchaseBlock)
+}
+
+// Todo(post-migration): make internal
+@objc(RCPurchasesOrchestrator) public class PurchasesOrchestrator: NSObject {
+
+    @objc public var finishTransactions: Bool { systemInfo.finishTransactions }
+    @objc public var allowSharingAppStoreAccount: Bool {
+        get {
+            if let allow = _allowSharingAppStoreAccount {
```

Comment:
could this be simplified to `_allowAppStoreSharingAccount ?? identityManager.currentUserIsAnonymous`?

Reply:
![nodding](https://user-images.githubusercontent.com/1304821/129789185-891b92c0-ff83-40a2-85a0-6904a9fcb0e0.gif)